### ISSUE TITLE
Request

### ISSUE THREAD

**Issue by LouisGarry** [bucanero/apollo-ps3#36]
Hi, hopefully the next version, to convert PS2, will be given 2 choices, ISO.BIN.ENC or title_id.BIN.ENC (sign _ not -).
Example: /dev_hdd0/PS2ISO/FFXII/SLUS_209.63.BIN.ENC

The reason is ISO.BIN.ENC does not support config from wMM.

Thank you.

**Comment by bucanero** [comment]
as a quick fix, I can change the behavior to create the ps2 classic file as `my_iso_original_filename.BIN.ENC`

so it would be:
`/dev_hdd0/PS2ISO/my_iso_original_filename.BIN.ENC`

detecting the title ID inside the ISO would require new code to be added and tested... but if you name your file properly, Apollo could just use the filename

**Comment by LouisGarry** [comment]
Hi master,  I see the title ID is on the offset 0x82A49-0x82A53 ISO game.

**Comment by bucanero** [comment]
> Hi master, I see the title ID is on the offset 0x82A49-0x82A53 ISO game.

that's valid for DVD .isos, but some ps2 games can be CD .isos, and in that case I think the offsets are different. If you can confirm with CD isos, that would help too.

**Comment by retrosapien** [comment]
I've always wanted a quick way to find the game id in a ps2 iso, so i tried to find it at that address in aggressive inline. I'm not that great with hex, but i know a bit. 
I couldn't find it at that address in this game. 
I did two searches, one for slus and another for the first 3 numbers, 203.

this is what i found;

BASLUS-20327CEAIGAME              3C1D48D0-3C1D48E3
BASLUS-20327IPARK                      3C1D4D58-3C1D4D69
SLUS_203.27;1                                82D9F-82DAB
BOOT2=cdrom0:\SLUS_203.27;       3C1E9800-3C1E9819

this is what i found at the offset LouisGarry mentioned.
����Ie 82A49

i also had a iso.bin.enc of that game, and i couldn't find either string at all. 
I don't know if any of those offsets are standard locations, or if this will help at all.
I just thought should share what i found.

**Comment by bucanero** [comment]
Some notes by @aldostools:

> IMHO it would be more convenient to use "my_file_name" + " [title id]" + ".BIN.ENC"
> Example: /dev_hdd0/PS2ISO/FFXII [SLUS_209.63].BIN.ENC
>
> This format would be more user friendly and would include the title id needed by webMAN MOD to find the CONFIG in the database.
>
> The title id can be obtained automatically reading the root directory of the ISO.

**Comment by retrosapien** [comment]
@aldostools
Will that format double the printing of the gameID in webman games on the xmb?
How do you read the gameid when the iso is encrypted? Without extracting anything?
The way mine is set up is;
/<Gameid><name>/iso.bin.enc
/<Gameid><name>/iso.bin.png
In webman games it appears as;
<name>/iso.bin [gameid]
Gameid | hdd0

I'd like it to have just the name with the gameid only in the subtitle. 
But I have never figured out how. 
(While also having the covers show and still work with placeholder, that is)

**Comment by retrosapien** [comment]
@aldostools
After some fiddling with settings and your advice as a stepping stool, I've gotten a bit closer. 
Using your example, except removing the brackets, I've gotten it to appear like this on the xmb;
name.bin
title id | hdd0
I had show id as part of name enabled in the settings. 
I tried to remove the .bin. part of the file name, and while it looked good on the xmb, it threw an error when trying to launc. 
Error path/filename.enc.config
Is that because the config file is not the same as the iso, now? Or will placeholder not support it?

**Comment by bucanero** [comment]
@retrosapien if you need support for webman mod, you'll have more chances if you post on psx-place.com forums, as I'm not sure if Aldos will be checking this messages here.

**Comment by retrosapien** [comment]
@bucanero 
Oh. No worries. I have already contacted @aldostools about that issue. 

Thanks for looking out. 
=)

For completeness ;
I ended up changing to his convention with a small change.
Using his example; 
Example: /dev_hdd0/PS2ISO/FFXII [SLUS_209.63].BIN.ENC
I changed to;
Example: /dev_hdd0/PS2ISO/SLUS_209.63.FFXII.BIN.ENC
If you leave the brackets on and/or put the titleid after the name, then the titleid shows as part of the file name on the xmb. 
I have the sfo option enabled in webman mod, so the titleid is already displayed as a subtitle under the game name.

**Comment by bucanero** [comment]
> as a quick fix, I can change the behavior to create the ps2 classic file as `my_iso_original_filename.BIN.ENC`
> 
> so it would be:
> `/dev_hdd0/PS2ISO/my_iso_original_filename.BIN.ENC`

this fix was added in 517dc4c157dd1ffd27c51d859cda7f2b311ab449 

Now in v1.6.0, PS2 classics are encrypted as `/dev_hdd0/PS2ISO/my-iso-original-filename.BIN.ENC`

**Comment by LouisGarry** [comment]
> > as a quick fix, I can change the behavior to create the ps2 classic file as `my_iso_original_filename.BIN.ENC`
> > so it would be:
> > `/dev_hdd0/PS2ISO/my_iso_original_filename.BIN.ENC`
> 
> this fix was added in [517dc4c](https://github.com/bucanero/apollo-ps3/commit/517dc4c157dd1ffd27c51d859cda7f2b311ab449)
> 
> Now in v1.6.0, PS2 classics are encrypted as `/dev_hdd0/PS2ISO/my-iso-original-filename.BIN.ENC`
@bucanero, hopefully the next update can include the title id as Aldostools meant
"my_file_name" + " [title id]" + ".BIN.ENC"
Example: /dev_hdd0/PS2ISO/FFXII [SLUS_209.63].BIN.ENC

Thanks.

**Comment by LouisGarry** [comment]
@bucanero, did you get inspiration for this request?

**Comment by bucanero** [comment]
> @bucanero, did you get inspiration for this request?

no, this is not going to happen (at least from my side). If someone wants to contribute and open a PR, I can review it.
